File: go/log.go
package log

import (
	"fmt"
	"os"
)

func Note(message any) {
	NoteColor.Fprintln(os.Stderr, message)
}

func Notef(format string, a ...any) {
	NoteColor.Fprintf(os.Stderr, format, a...)
	fmt.Fprintln(os.Stderr)
}

func Success(message any) {
	SuccessColor.Fprintln(os.Stderr, message)
}

func Successf(format string, a ...any) {
	SuccessColor.Fprintf(os.Stderr, format, a...)
	fmt.Fprintln(os.Stderr)
}

func Warn(message any) {
	WarnColor.Fprintln(os.Stderr, message)
}

func Warnf(format string, a ...any) {
	WarnColor.Fprintf(os.Stderr, format, a...)
	fmt.Fprintln(os.Stderr)
}

func Error(message any) {
	ErrorColor.Fprintln(os.Stderr, message)
}

func Errorf(format string, a ...any) {
	ErrorColor.Fprintf(os.Stderr, format, a...)
	fmt.Fprintln(os.Stderr)
}

func Fatal(message any) {
	FatalColor.Fprintln(os.Stderr, message)
	os.Exit(1)
}

func Fatalf(format string, a ...any) {
	FatalColor.Fprintf(os.Stderr, format, a...)
	fmt.Fprintln(os.Stderr)
	os.Exit(1)
}

File: zig/log.zig
const std = @import("std");

pub const note_color = "\x1B[2m";
pub const warn_color = "\x1B[33m";
pub const error_color = "\x1B[31m";
pub const fatal_color = "\x1B[31;1m";

const reset = "\x1B[0m";

pub fn log(comptime color: []const u8, comptime format: []const u8, args: anytype) void {
    std.debug.print(color ++ format ++ reset ++ "\n", args);
}

pub fn note(comptime format: []const u8, args: anytype) void {
    log(note_color, format, args);
}
pub fn warn(comptime format: []const u8, args: anytype) void {
    log(warn_color, format, args);
}
pub fn err(comptime format: []const u8, args: anytype) void {
    log(error_color, format, args);
}
pub fn fatal(comptime format: []const u8, args: anytype) noreturn {
    log(fatal_color, format, args);
    std.process.exit(1);
}
